tests/utils/test_file_utils.py

```diff
@@ -364,3 +371,106 @@ def test_shutil_copytree_without_file_permissions(tmp_path):
     assert set(os.listdir(dst_dir.joinpath("subdir"))) == {"subdir-file.txt"}
     assert dst_dir.joinpath("subdir/subdir-file.txt").read_text() == "testing 123"
     assert dst_dir.joinpath("top-level-file.txt").read_text() == "hi"
+
+
+def test_parallelized_download_file_using_http_uri_requests_appropriate_chunks():
+    calls_kwargs = multiprocessing.Manager().list([])
+
+    # Get call kwargs manually. Calls are not properly stored in a MagicMock object
+    # in a multiprocessing context.
+    def mock_request_side_effect(method, url, *args, **kwargs):
+        calls_kwargs.append(kwargs)
+        response_mock = Response()
+        response_mock.status_code = 206
+        response_mock._content = b"\x01\x01"
+        return response_mock
+
+    # Workaround for windows preventing re-opening of an open handle
+    f = tempfile.NamedTemporaryFile(delete=False)
+    f.close()
+    with mock.patch("requests.Session.request") as request_mock:
+        request_mock.side_effect = mock_request_side_effect
+        parallelized_download_file_using_http_uri(
+            "fake_uri",
+            f.name,
+            file_size=1000,
+            uri_type=ArtifactCredentialType.AWS_PRESIGNED_URL,
+            chunk_size=100,
+            headers={},
+        )
+    f.close()
+    requested_ranges = [call_kwargs["headers"]["Range"] for call_kwargs in calls_kwargs]
+    assert len(requested_ranges) == 10
+    expected_ranges = []
+    for i in range(10):
+        byte_range = f"bytes={100*i}-"
+        if i != 9:
+            byte_range += f"{(100*(i+1))-1}"
+        expected_ranges.append(byte_range)
+    assert set(requested_ranges) == set(expected_ranges)
```

Comment:
```suggestion
    assert sorted(requested_ranges) == expected_ranges
```

`sorted` might be better here because `set` hides duplicates.

Reply:
Nice catch :)